Makefile

```diff
@@ -59,163 +34,230 @@ ifeq ("$(DEBUG_BUILD)", "true")
 	GO_BUILD = $(GO_CMD) build -gcflags='all=-N -l'
 endif
 
-# To build a specific image, run 'make <name>-image' (e.g. 'make postgres-operator-image')
-images = postgres-operator \
-	crunchy-postgres-exporter
-
-.PHONY: all setup clean push pull release deploy
-
-
-#======= Main functions =======
-all: $(images:%=%-image)
+##@ General
+
+# The help target prints out all targets with their descriptions organized
+# beneath their categories. The categories are represented by '##@' and the
+# target descriptions by '##'. The awk command is responsible for reading the
+# entire set of makefiles included in this invocation, looking for lines of the
+# file as xyz: ## something, and then pretty-formatting the target and help. Then,
+# if there's a line with ##@ something, that gets pretty-printed as a category.
+# More info on the usage of ANSI control characters for terminal formatting:
+# https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+# More info on the awk command:
+# http://linuxcommand.org/lc3_adv_awk.php
+
+.PHONY: help
+help: ## Display this help.
```

Comment:
👍🏻 I like that `help` is the first target. That way, it runs when `make` is called without any arguments.

📝 I'll take a look after this merges to see if it improves the behavior of our CodeQL workflow. (I see `buildah` running in the 4-minute-long [Autobuild step here](https://github.com/CrunchyData/postgres-operator/actions/runs/3886078304/jobs/6630751105#step:5:149).)

https://github.com/CrunchyData/postgres-operator/blob/a1397fde1e4d5351186840f65bca23a572f3a51f/.github/workflows/codeql-analysis.yaml#L33-L37

Reply:
Ohh yeah that'll be nice if it does speed things up

Reply:
Autobuild is maybe twice as fast now. 👍🏻 

Before: [Job 6745309870](https://github.com/CrunchyData/postgres-operator/actions/runs/3942085391/jobs/6745309870)
After: [Job 6862717663](https://github.com/CrunchyData/postgres-operator/actions/runs/3999162447/jobs/6862717663)